### PR TITLE
Proper computation of realtime "segment.flush.threshold.size" in case of force-commit

### PR DESCRIPTION
**Pull request by sajjad-moradi**
The `SegmentFlushThresholdComputer` does not appropriately manage situations where segments are committing as a result of force-commit requests. In this PR, the size threshold of the committing segment is selected as the value for the new consuming segment.

### CODE REVIEW

File: pinot-controller/src/test/java/org/apache/pinot/controller/helix/core/realtime/segment/SegmentFlushThresholdComputerTest.java [hunk 1]
```diff
@@ -105,6 +106,28 @@ public void testUseLastSegmentsThresholdIfSegmentSizeMissing() {
...
 
+  @Test
+  public void testUseLastSegmentsThresholdIfSegmentIsCommittingDueToForceCommit() {
+    long committingSegmentSizeBytes = 500_000L;
+    int committingSegmentSizeThreshold = 25_000;
+    SegmentFlushThresholdComputer computer = new SegmentFlushThresholdComputer();
+
+    CommittingSegmentDescriptor committingSegmentDescriptor = mock(CommittingSegmentDescriptor.class);
+    when(committingSegmentDescriptor.getSegmentSizeBytes()).thenReturn(committingSegmentSizeBytes);
+    when(committingSegmentDescriptor.getStopReason()).thenReturn(REASON_FORCE_COMMIT_MESSAGE_RECEIVED);
```
Comment (mcvsubbu) [inline pinot-controller/src/test/java/org/apache/pinot/controller/helix/core/realtime/segment/SegmentFlushThresholdComputerTest.java:117]: maybe also add a test when the reason is null. (will happen during upgrades if servers are upgraded last)

**Review by jtao15 — APPROVED** [review]
LGTM, thanks for fixing this!

File: pinot-controller/src/test/java/org/apache/pinot/controller/helix/core/realtime/segment/SegmentFlushThresholdComputerTest.java (hunk 1 above)
Comment (sajjad-moradi) [reply pinot-controller/src/test/java/org/apache/pinot/controller/helix/core/realtime/segment/SegmentFlushThresholdComputerTest.java:117]: I have actually considered the upgrade scenario, and I believe a unit test is unnecessary in this case. This is because the behavior during the upgrade will be managed in the same way as it was before the upgrade, with the threshold being calculated using much lower values. To put it differently, the miscalculation bug is present both before and during the upgrade. However, after the upgrade, the bug is resolved.

**Review by Jackie-Jiang — APPROVED** [review]

**Review by mcvsubbu — APPROVED** [review]
lgtm, thanks
